### PR TITLE
AWS Network Manager Resource w/ Core & Global 

### LINKED ISSUE #9511: Add AWS NetworkManager Resource Type
### Describe the feature

Add AWS NetworkManager Resource Type
https://docs.aws.amazon.com/networkmanager/latest/APIReference/Welcome.html

### Extra information or context

_No response_

### PR DESCRIPTION
**Pull request by zkarpinski**
Adds support for AWS Network Manager resource. Including Core Network and Global Network.
Closes #9511 #9512 & #9513 

**GlobalNetwork**

- DescribeGlobalNetworks
- TagResource
- UntagResource
- DeleteGlobalNetwork

**Core Network**

- ListCoreNetworks
- GetCoreNetwork
- TagResource
- UntagResource
- DeleteCoreNetwork

### CODE REVIEW

**Comment by zkarpinski** [comment]
@kapilt @PratMis Not sure what I'm missing that is causing the failure 
`"`_FAILED tests/test_policy.py::PolicyMetaLint::test_resource_name - AssertionError: networkmanager-core-network, networkmanager-global-network dont have resource name for reporting_`"`

File: c7n/resources/networkmanager.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            'get_core_network', 'CoreNetworkId',
+            'CoreNetworkId', None)
+        arn = 'CoreNetworkArn'
+        name = 'CoreNetworkId'
+        id = 'CoreNetworkId'
+        date = 'CreatedAt'
+        config_type = None
+        cfn_type = 'AWS::NetworkManager::CoreNetwork'
+        permissions_augment = ("networkmanager:ListTagsForResource",)
+        universal_taggable = object()
```
Comment (jerryhxu) [inline c7n/resources/networkmanager.py:41]: No need for this line since 'tag' and 'untag' are implemented.

File: c7n/resources/networkmanager.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    class resource_type(TypeInfo):
+        service = 'networkmanager'
+        enum_spec = ('describe_global_networks', 'GlobalNetworks', None)
+        arn = 'GlobalNetworkArn'
+        name = 'GlobalNetworkId'
+        id = 'GlobalNetworkId'
+        date = 'CreatedAt'
+        config_type = cfn_type = 'AWS::NetworkManager::GlobalNetwork'
+        permissions_augment = ("networkmanager:ListTagsForResource",)
+        universal_taggable = object()
```
Comment (jerryhxu) [inline c7n/resources/networkmanager.py:58]: No need for this line since 'tag' and 'untag' are implemented.

File: c7n/resources/networkmanager.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+    source_mapping = {'describe': DescribeGlobalNetwork, 'config': ConfigSource}
```
Comment (jerryhxu) [inline c7n/resources/networkmanager.py:60]: No need for this line since it doesn't provide any additional functionality.

**Review by jerryhxu — CHANGES_REQUESTED** [review]
Please also implement mark-for-op and marked-for-op as part of tagging functionality.

File: c7n/resources/networkmanager.py (hunk 3 above)
Comment (zkarpinski) [reply c7n/resources/networkmanager.py:60]: Hi @jerryhxu , If I remove this line, I get the following error message.
`AttributeError: 'NetworkManager' object has no attribute 'get_global_network'. Did you mean: 'create_global_network'?`

File: c7n/resources/networkmanager.py (hunk 1 above)
Comment (zkarpinski) [reply c7n/resources/networkmanager.py]: Removed. Thanks!

File: c7n/resources/networkmanager.py (hunk 2 above)
Comment (zkarpinski) [reply c7n/resources/networkmanager.py]: Removed. Thanks!

File: c7n/resources/networkmanager.py (hunk 3 above)
Comment (zkarpinski) [reply c7n/resources/networkmanager.py]: Removed. Thanks!

File: c7n/resources/networkmanager.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+            'CoreNetworkId', None)
+        arn = 'CoreNetworkArn'
+        name = 'CoreNetworkId'
+        id = 'CoreNetworkId'
+        date = 'CreatedAt'
+        config_type = None
+        cfn_type = 'AWS::NetworkManager::CoreNetwork'
+        permissions_augment = ("networkmanager:ListTagsForResource",)
+
+    source_mapping = {'describe': GetCoreNetwork, 'config': ConfigSource}
```
Comment (jerryhxu) [inline c7n/resources/networkmanager.py:42]: This line is probably not needed. Both class GetCoreNetwork and DescribeGlobalNetwork don't seem to provide anything useful. They probably can be removed as well.

**Review by jerryhxu — CHANGES_REQUESTED** [review]

File: c7n/resources/networkmanager.py (hunk 4 above)
Comment (zkarpinski) [reply c7n/resources/networkmanager.py]: Thanks! Change made

File: c7n/resources/networkmanager.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+from c7n.actions.core import BaseAction
+from c7n.manager import resources as c7n_resources
+from c7n.query import QueryResourceManager, TypeInfo, DescribeSource
+from c7n.utils import local_session, type_schema
+from c7n.tags import RemoveTag, Tag, TagActionFilter, TagDelayedAction
+
+
+class GetCoreNetwork(DescribeSource):
```
Comment (jerryhxu) [inline c7n/resources/networkmanager.py:11]: This class has no use. Probably can be deleted

File: c7n/resources/networkmanager.py [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    schema = type_schema('delete')
+    permissions = ('networkmanager:DeleteGlobalNetwork',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('networkmanager')
+
+        for r in resources:
+            try:
+                client.delete_global_network(GlobalNetworkId=r['GlobalNetworkId'])
+            except client.exceptions.ResourceNotFound:
```
Comment (jerryhxu) [inline c7n/resources/networkmanager.py:147]: Should be ResourceNotFoundException

File: c7n/resources/networkmanager.py [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+    schema = type_schema('delete')
+    permissions = ('networkmanager:DeleteCoreNetwork',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('networkmanager')
+
+        for r in resources:
+            try:
+                client.delete_core_network(CoreNetworkId=r['CoreNetworkId'])
+            except client.exceptions.ResourceNotFound:
```
Comment (jerryhxu) [inline c7n/resources/networkmanager.py:130]: Should be ResourceNotFoundException

**Review by jerryhxu — CHANGES_REQUESTED** [review]

File: c7n/resources/networkmanager.py [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+
+
+class GetCoreNetwork(DescribeSource):
+
+    def augment(self, resources):
+        resources = super().augment(resources)
+        return resources
+
+
+class DescribeGlobalNetwork(DescribeSource):
```
Comment (jerryhxu) [inline c7n/resources/networkmanager.py:18]: This class has no use. Probably can be deleted

**Review by jerryhxu — CHANGES_REQUESTED** [review]

**Review by thisisshi — APPROVED** [review]

**Pull request merged**